packages/x-adapter/src/adapter.types.ts

```diff
@@ -0,0 +1,70 @@
+import { HttpClient, RequestOptions } from './http-client.types';
+import { Mapper } from './mapper.types';
+
+/**
+ * A facade containing all the different {@link EndpointAdapter} methods.
+ *
+ * @public
+ */
+export interface Adapter {}
```

Comment:
I've intentionally left this interface empty, so we can add the requests and responses later one by one.